test/drivers/s3.test.ts

```diff
@@ -12,15 +12,16 @@ const region = process.env.VITE_S3_REGION;
 describe.skipIf(
   !accessKeyId || !secretAccessKey || !bucket || !endpoint || !region
 )("drivers: s3", () => {
+  const driver = s3Driver({
```

Comment:
for whatever reason, vitest does actually call this overall callback even if the `skipIf` is true (i.e. if it is skipped)

so this still needs to be:

```suggestion
  const driver = () => s3Driver({
```

since, if `endpoint` isn't set, it'll throw even if this test is skipped